### PR TITLE
Oded/per 9900 net sdk request from fcma

### PR DESCRIPTION
**Pull request by obsd**

### CODE REVIEW

File: tests/PermitTests/PermitClientTests.cs [hunk 1]
```diff
@@ -103,12 +120,8 @@ public async void TestPermitClientApi()
             Assert.True(assignedRoles.Count == (assignedRoles2.Count - 1));
 
             await permitClient.Api.DeleteUser(getUser.Key);
-            getUser = await permitClient.Api.GetUser(user.Key);
-            Assert.True(getUser == null);
-            
+
```
Comment (omer9564) [inline tests/PermitTests/PermitClientTests.cs:123]: why did you delete this ?

File: tests/PermitTests/PermitClientTests.cs [hunk 2]
```diff
@@ -103,12 +120,8 @@ public async void TestPermitClientApi()
...
             await permitClient.Api.DeleteTenant(tenantKey);
-            getTenant = await permitClient.Api.GetTenant(getTenant.Id.ToString());
```
Comment (omer9564) [inline tests/PermitTests/PermitClientTests.cs:110]: why did you delete this ?

File: tests/PermitTests/PermitClientTests.cs [hunk 3]
```diff
@@ -55,34 +57,49 @@ public async void TestPermitClientApi()
...
+            var resourceInstanceKey = String.Concat("testResourceInstance11111111111111", postfix);
+            var resourceInstance = await permitClient.Api.CreateResourceInstance(
+                new ResourceInstanceCreate { Key = resourceInstanceKey, Resource = resourceKey, Tenant = tenantKey }
+            );
+            var getResourceInstance = await permitClient.Api.GetResourceInstance(resourceInstance.Id.ToString());
+            Assert.True(resourceInstance.Key == getResourceInstance.Key);
+
             // test roles Api
-            string roleName = "rName";
+            string roleName = String.Concat("rName11111111111111", postfix);
```
Comment (omer9564) [inline tests/PermitTests/PermitClientTests.cs:96]: why change the name here ?

File: tests/PermitTests/PermitClientTests.cs [hunk 4]
```diff
@@ -55,34 +57,49 @@ public async void TestPermitClientApi()
...
-            
-            var getTenant2 = await permitClient.Api.UpdateTenant(tenantKey, new TenantUpdate {Description = desc2});
+
+            var getTenant2 = await permitClient.Api.UpdateTenant(tenantKey, new TenantUpdate { Description = desc2 });
             Assert.True(getTenant2.Description == desc2);
             Assert.True(getTenant.Key == tenantKey);
 
+
+            // test Resource Api
+            var resourceKey = String.Concat("testResource11111111111111", postfix);
```
Comment (omer9564) [inline tests/PermitTests/PermitClientTests.cs:82]: why change the name like this?

File: tests/PermitTests/PermitClientTests.cs [hunk 5]
```diff
@@ -55,34 +57,49 @@ public async void TestPermitClientApi()
...
             Assert.True(getUser.Email == testEmail);
 
             // test Tenant Api
             //string desc = "desc";
-            string desc2 = "desc2";
-            string tenantName = "tName";
-            string tenantKey = "tKey";
-            var tenantCreate = new TenantCreate {Key = tenantKey, Name = tenantName};
+            string desc2 = "desc21";
+            string tenantName = "tName11111111111111";
```
Comment (omer9564) [inline tests/PermitTests/PermitClientTests.cs:69]: same comment about naming here

**Review by omer9564 — APPROVED** [review]
LGTM, only had comments about the string naming on the tests, if you want them to be unique add a uuid or some random string to it

File: tests/PermitTests/PermitClientTests.cs (hunk 1 above)
Comment (obsd) [reply tests/PermitTests/PermitClientTests.cs:125]: This test isn't working, you delete it and get 404

File: tests/PermitTests/PermitClientTests.cs (hunk 2 above)
Comment (obsd) [reply tests/PermitTests/PermitClientTests.cs:110]: You cannot get after delete you get 404 exception
